hub-saml/src/main/java/uk/gov/ida/saml/core/validators/subject/AssertionSubjectValidator.java

```diff
@@ -26,7 +29,11 @@ public void validate(
             throw new SamlTransformationErrorException(failure.getErrorMessage(), failure.getLogLevel());
         }
 
-        if (!NameIDType.PERSISTENT.equals(subject.getNameID().getFormat())) {
+        boolean correctNameIdType = Stream
+                .of(NameIDType.PERSISTENT, NameIDType.TRANSIENT)
+                .anyMatch(type -> type.equalsIgnoreCase(subject.getNameID().getFormat()));
+
```

Comment:
Should we ignore case here? URIs are case sensitive and we've pushed back on this before. Do we need this for what DE are sending?